Gordon360/Static Classes/Names.cs

```diff
@@ -121,17 +121,19 @@ public static class SQLQuery
         //ALL_ALUMNI_REQUEST returns the set of all the alumni in the records
         //Other methods query this set to find a specific alum based on AD_Username (or the encrypted id)
         public static string ALL_ALUMNI_REQUEST = "SELECT ID, WebUpdate, Title, FirstName, MiddleName, LastName, Suffix, MaidenName, NickName, HomeStreet1, HomeStreet2, HomeCity, HomeState, HomePostalCode, HomeCountry, HomePhone, HomeFax, HomeEmail, JobTitle, MaritalStatus, SpouseName, College, ClassYear, PreferredClassYear, Major1, Major2, ShareName, ShareAddress, Gender, GradDate, Email, grad_student, Barcode, ISNULL(AD_Username, CONVERT(NVARCHAR(32),HashBytes('SHA2_256', ID),2)) as AD_Username, show_pic, preferred_photo, Country, Major2Description, Major1Description from Alumni";// WHERE AD_Username is not null";
-        public static string ALL_BASIC_INFO_NOT_ALUM = 
-            "select DISTINCT A.firstname as Firstname, A.lastname as Lastname, " +
-            "COALESCE(B.NickName, C.Nickname) as Nickname, " +
-            "COALESCE(B.MaidenName, C.MaidenName) as MaidenName, A.AD_Username as Username, " +
+        public static string ALL_BASIC_INFO_NOT_ALUM =
+            "select A.firstname, A.lastname, S.Nickname, S.MaidenName, A.AD_Username as Username, " +
             "CONCAT(A.firstname, A.lastname, A.AD_Username) as ConcatonatedInfo " +
-            "from ACCOUNT as A left join FacStaff as B on A.AD_Username = B.AD_Username " +
-            "left join Student as C on A.AD_Username = C.AD_Username " +
-            "where isnull(A.AD_Username, '') != '' " +
-            "and(isnull(B.AD_Username, '') != '' " +
-            "or isnull(C.AD_Username, '') !='')";
-            
+            "from Student as S " +
+            "left join ACCOUNT as A on A.AD_Username = S.AD_Username where A.AD_Username is not null " +
+            "and S.AD_Username is not null " +
+            "union " +
+            "select A.firstname, A.lastname, F.NickName, F.MaidenName, A.AD_Username as Username, " +
+            "CONCAT(A.firstname, A.lastname, A.AD_Username) as ConcatonatedInfo " +
+            "from FacStaff as F " +
+            "left join ACCOUNT as A on F.AD_Username = A.AD_Username where A.AD_Username is not null " +
+            "and F.AD_Username is not null";
+
```

Comment:
I think this union syntax is very clear and should be equally performant! Two notes though:

1. Using `left join` and then checking that the join condition is not null achieves the same behavior as an inner join, but in a roundabout way. These should just be inner joins to `ACCOUNT`.
2. Was there a reason to make this a string rather than editing the existing stored procedure? Having a whole query in a string is a VERY easy way to get hard-to-notice bugs because there is absolutely 0 syntax verification/error-checking on your query in this format. Also, stored procedures tend to have some run-off performance benefits because SQL Server can analyze the query ahead of time and keep certain things cached.

Reply:
We can write a decently exhaustive pros and cons list for this:

### Stored Procedure:

Pros
- Cached in database for performance, consistency
- Ease of editing - doesn't require a redeployment to be updated
- syntax verification & error checking - SQL prevents you from saving a stored procedure that won't work for 95% of the most common reasons
- Centralization of complex database behavior

Cons
- Separate from the codebase. Have to be familiar with SQL tools to find, much less edit, it
- Ease of editing - not version controlled beyond manually logging updates. No way to revert changes.

### String in API 

Pros
- Properly version controlled - commit history of all changes, enabling reverts, tracking changes and updates historically, and all the other advantages of version control
- Findable within the codebase - don't have to leave the API code to find it

Cons
- Just a string - no syntax verification, no error checking, no sanity checks at all. C# lets you put anything you want in the string, no matter how massively or obscurely broken it may be
- Less performant - C# has to parse the string into SQL at runtime, and then send it to the SQL server which has to turn the declarative query into an execution plan on every invocation of the procedure
- Not able to be updated alongside the DB - if something in the database has to change, we have to update the database and simultaneously (as much as is possible) deploy new API code. This makes things like testing and cascading updates very difficult

Performance is not a major issue at this point in time, but the other two cons of storing queries as strings in the API massively outweigh the pros, in my opinion. The lack of syntax, error, and sanity checking has resulted in MANY issues in the past. And it will continue to be an issue every time someone works on these queries, much less tries to create new ones.

Furthermore, the advantages of version control unfortunately almost work against us in this case, because the rest of our database is not version controlled. If this were a code-first database where all the database structure was defined in EF code and could be version controlled through migrations, then that would be the end of the discussion right there. However, because that is not how 360 is built, it is more complicated. For example - say we needed to change the ACCOUNT table in the database for some reason. If the query is a stored procedure, we can write on SQL script that will update the ACCOUNT table and the query that depends on it in the same transaction, preventing any possibility of inconsistency. However, if the query is a string in the API, we have to micromanage the changes to make sure that DB changes happen at the same time as deployments. This also means that to test any DB changes, we would have to break Train until the changes are tested and pushed.

For those reasons, unless we can find a way of encoding the query such that it has syntax and error checking against the EF table definitions, I can't condone adding new stringly-typed queries to the API.

Reply:
Thanks for this pro and con, it is really helpful to understand the differences in both ways.

I just made another commit that changed the SQL back to what it should be, so the modification will only take place in the databse.

Reply:
I'm glad to hear it was helpful! I appreciate y'all paying attention to the finer details on quality and maintainability for our codebase, rather than just settling for "it works well enough".

Since the only change now is the added comments, is it still worth merging this pr? I guess the comments don't hurt anything.

Reply:
It is the same on 360. But since I merged a PR with modified SQL to develop, we probably should merge this to develop again before the release this afternoon.